spring-cloud-stream-configuration-metadata/src/main/java/org/springframework/cloud/stream/configuration/metadata/ModuleConfigurationMetadataResolver.java

```diff
@@ -45,6 +46,46 @@ public ModuleConfigurationMetadataResolver() {
 	}
 
 	/**
+	 * Return metadata about configuration properties (as groups) that are documented via
+	 * <a href="http://docs.spring.io/spring-boot/docs/current/reference/html/configuration-metadata.html">
+	 * Spring Boot configuration metadata</a> and visible in a module.
+	 *
+	 * @param module a Spring Cloud Stream module; typically a Boot uberjar,
+	 * but directories are supported as well
+	 */
+	public List<ConfigurationMetadataGroup> listPropertyGroups(Resource module) {
```

Comment:
@ericbottard Could you help me understand how are the methods in this class invoked?